### PR TITLE
Code Quality: Move platform-depends classes from Shared to Core

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Code Quality: Move platform-depends classes from Shared to Core

cc: @d2dyno1 

### Motivation & Context

- Improved readability

> **Warning**<br/>Please do not review code lines that I did not changed. I cannot make more diff. If there're more codes that are preferred to change, I will do that and mark that PR as the final one for that stage.

### PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Related #4180
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

### Screenshots

image

### CODE REVIEW

**Comment by d2dyno1** [comment]
Looks like the pipeline is failing

File: src/Files.Core/Utils/Cloud/CloudProviders.cs [hunk 1]
```diff
@@ -1,27 +1,44 @@
 ﻿// Copyright (c) 2023 Files Community
 // Licensed under the MIT License. See the LICENSE.
 
-namespace Files.Shared.Cloud
+namespace Files.Core.Utils.Cloud
 {
 	public enum CloudProviders
 	{
 		OneDrive,
+
```
Comment (yaira2) [inline src/Files.Core/Utils/Cloud/CloudProviders.cs:9]: We don't usually add empty lines between enums

Comment (ferrariofilippo) [reply src/Files.Core/Utils/Cloud/CloudProviders.cs:9]: I think they're planning to add comments there

Comment (0x5bfa) [reply src/Files.Core/Utils/Cloud/CloudProviders.cs:9]: Yes. However, I cannot afford to add comments.

Comment (yaira2) [reply src/Files.Core/Utils/Cloud/CloudProviders.cs:9]: @0x5bfa can you remove the empty lines for now? We can add them back once comments are added.

Comment (0x5bfa) [reply src/Files.Core/Utils/Cloud/CloudProviders.cs:9]: Added comments instead.

**Review by d2dyno1 — APPROVED** [review]
Looks good!

**Comment by yaira2** [comment]
@0x5bfa can you resolve the merge conflicts?

**Comment by 0x5bfa** [comment]
Fixed conflicts

**Pull request merged**